src/TheThingsNetwork.cpp

```diff
@@ -201,16 +201,15 @@ bool TheThingsNetwork::enableFsbChannels(int fsb) {
   return true;
 }
 
-bool TheThingsNetwork::personalize(const byte devAddr[4], const byte nwkSKey[16], const byte appSKey[16], bool reset) {
+bool TheThingsNetwork::personalize(const byte devAddr[4], const byte nwkSKey[16], const byte appSKey[16]) {
   
-  if (reset) {
+  if (devAddr != NULL && nwkSKey != NULL && appSKey != NULL) {
```

Comment:
@johanstokking would know better, but I think it's preferable to have a separate proto with no args. That makes it more clear that you either need to pass all 3 or none.